manifests/profile/solr.pp

```diff
@@ -0,0 +1,117 @@
+# Copyright (c) 2018-2019 The Regents of the University of Michigan.
+# All Rights Reserved. Licensed according to the terms of the Revised
+# BSD License. See LICENSE.txt for details.
+
+# nebula::profile::solr
+#
+# Provisions a single instance of solr.
+# For each given "shard", this provisions the lib and conf directory 
+# for an "x" and "y" core and symlinks the data directory for the shard.
+#
+# @param timezone  The timezone to use for logs, etc. Should normally be set via Hiera.
+# @param coredata  A map of core names to the full paths on disk for the data
+#                  directory for those cores.
+# @param base   The directory under which all solr files will be deployed
+# @param home   The value to use for HOME; solr cores go under here
+# @param logs   The directory to use LOGS_DIR; where all the logging goes
+# @param port   The port solr should listen on
+# @param heap   How much memory to allocate for the Solr heap.
+#
+# @example
+#   class { 'nebula::profile::solr':
+#     timezone                 => 'America/New_York'
+#     coredata                 => {
+#                      'core1' => '/path/to/core_1/data',
+#                      'core2' => '/path/to/core_2/data'
+#                   }
+#   }
+class nebula::profile::solr (
+  String $timezone,
+  Hash[String,String] $coredata = {},
+  String $base = '/var/lib/solr',
+  String $home = "${base}/home",
+  String $logs = "${base}/logs",
+  String $heap = '32G',
```

Comment:
Let's start with 8G here.

Reply:
It looks like none of our app-supporting general solrs are above 3G in production (search solr is an outlier at 8G), and that's with some of them hosting multiple cores. Even the solr backing the library web site Drupal is only 1G.  I'd recommend 1G as the default for minimal needs and we increase via hiera as indicated.

Reply:
Done.